gdal/frmts/grass/grass57dataset.cpp

```diff
@@ -464,8 +456,19 @@ CPLErr GRASSRasterBand::ResetReading ( struct Cell_head *sNewWindow )
              sNewWindow->rows   != sCurrentWindow.rows   || sNewWindow->cols   != sCurrentWindow.cols
              )
         {
+            if( hCell >= 0 ) {
```

Comment:
as those same lines seem to be used at line 431 too, maybe it would make sense to move them to a dedicated method ?

Reply:
It's only four lines of code that appear only twice, and only in `GRASSRasterBand::ResetReading`.
Maybe move them to the top of `GRASSRasterBand::ResetReading` to avoid code duplication? The actual data reading functions `GRASSRasterBand::IReadBlock` and `GRASSRasterBand::IRasterIO` now always open the GRASS raster if not yet opened.

Reply:
> Maybe move them to the top of `GRASSRasterBand::ResetReading` to avoid code duplication?

if that's doable, yes, that would be better for future maintainance